.github/workflows/submitty_ci_2.yml

```diff
@@ -0,0 +1,216 @@
+              key: ${{ runner.os }}-py-pip-${{ hashFiles('**/requirements.txt') }}
+              restore-keys: |
+                ${{ runner.os }}-py-pip
+          - name: Install python dependencies
+            run: |
+              python3 -m pip install --upgrade pip setuptools wheel  
+              sudo python3 -m pip install --upgrade pip setuptools wheel
+              pip3 -V
+              sudo pip3 -V
+              pip3 install cryptography
+              pip3 install python-pam
+              pip3 install sqlalchemy
+              pip3 install selenium
+              pip3 install tzlocal
+              pip3 install jsonschema
+              pip3 install jsonref
+              pip3 install websocket_client
+              pip3 install psycopg2
+              pip3 install docker
+              pip3 install pyyaml
+              pip3 install selenium
+              pip3 install websocket_client
+              pip3 install python-pam
+              pip3 install python-dateutil
+              pip3 install paramiko
+              pip3 install pyzbar
+              pip3 install watchdog
+              #needs to be globally available for all users
+              sudo pip3 install docker
+              sudo pip3 install paramiko
+              sudo pip3 install tzlocal
+              sudo pip3 install sqlalchemy
+              sudo pip3 install jsonref
+              sudo pip3 install numpy
+              sudo pip3 install opencv-python
+              sudo pip3 install pyPdf
+              sudo apt-get install libzbar0
+              sudo pip3 install pyzbar
+              sudo pip3 install PyPDF2
+              sudo pip3 install pdf2image
+              sudo pip3 install onnxruntime
+              sudo pip3 install watchdog
+              sudo pip3 install python-dateutil
+              sudo pip3 install psutil
+              sudo pip3 install psycopg2
+              sudo pip3 install python-pam
+              sudo pip3 install sqlalchemy
+          - name: Get composer cache dir
+            id: composer-cache
+            run: |
+              pushd ${SUBMITTY_INSTALL_DIR}/GIT_CHECKOUT/Submitty/site
+              echo "::set-output name=dir::$(composer config cache-files-dir)"
+              popd
+          - name: Install composer Cache
+            uses: actions/cache@v2
+            with:
+              path: ${{ steps.composer-cache.outputs.dir }}
+              key: ${{ runner.os }}-php-composer-${{ hashFiles('**/composer.lock') }}
+              restore-keys: |
+                ${{ runner.os }}-php-composer-
+          - name: Install PHP dependencies
+            run: |
+              pushd ${SUBMITTY_INSTALL_DIR}/GIT_CHECKOUT/Submitty/site
+              composer install --prefer-dist
+              popd
+          - name: Install Submitty python utils
+            run: |
+              export SUBMITTY_INSTALL_DIR=/usr/local/submitty 
+              export SUBMITTY_REPOSITORY=${SUBMITTY_INSTALL_DIR}/GIT_CHECKOUT/Submitty
```

Comment:
I'd rather see these exports just moved into the `env` map at the top of the file, with maybe a comment saying these variables should have same value as `SUBMITTY_INSTALL_DIR`, like:

```
env:
    # ...
    SUBMITTY_INSTALL_DIR: /usr/local/submitty
    # The following variables should all use a stem that matches SUBMITTY_INSTALL_DIR above
    SUBMITTY_REPOSITORY: /usr/local/submitty/GIT_CHECKOUT/Submitty
```

Otherwise, if we update one `SUBMITTY_REPOSITORY` definition, we've got to update multiple, and DRY.

Reply:
Yeah the `SUBMITTY_REPOSITORY` should be a top level `env` variable, I remember running into an issue where some of the bash scripts couldn't see those env variables unless they were defined with an export locally in the same run step, but that could have been a fluke from something else? I'll have to remove and re-test it